File: swift/392_is_subsequence.swift
class Solution {
    /// isSubsequence returns true if s is a suvsequence of t, or false otherwise
    func isSubsequence(_ s: String, _ t: String) -> Bool {
        var currentIndex: String.Index = t.startIndex
        
        for char in s {
            while currentIndex < t.endIndex && char != t[currentIndex] {
                currentIndex = t.index(after: currentIndex)
            }
            
            if currentIndex == t.endIndex {
                return false
            }
            
            currentIndex = t.index(after: currentIndex)
        }
        
        return true
    }
}

File: python/392_is_subsequence.py
class Solution:
    ### isSubsequence returns true if s is a suvsequence of t, or false otherwise
    def isSubsequence(self, s: str, t: str) -> bool:
        currentIndex = 0
        
        for char in s:
            while currentIndex < len(t) and char != t[currentIndex]:
                currentIndex += 1
            
            if currentIndex == len(t):
                return False
            
            currentIndex += 1
        
        return True
        